### ISSUE TITLE
Python3: watchdog cannot patch _ctypes.COMError's constructor 

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/nvda#9763]
### Steps to reproduce:
When running NVDA from the threshold_py3_staging branch, it fails to start as watchdog cannot replace _ctypes.COMError's constructor with its own custom version in order to raise CallCanceled.
 
### Actual behavior:
The following exception is raised:
```
TypeError: can't set attributes of built-in/extension type '_ctypes.COMError'
```

### Expected behavior:
Setting the attribute should succeed.

### More details
Watchdog needs to override _ctypes.COMError's constructor so that it can raise our own custom CallCanceled exception when ever comtypes detects a COMError. This is so our code can tell the difference between a canceled call, and any other COMError. There are many many places in our code where we catch COMError, and it would be a lot of work to also check for CallCanceled and re-raise., thus why we override COMError's constructor.
In Python2, COMError had replacable attributes, but in Python3, the __dict__ and __slots__ seems to be now readonly.
One thing I tried was:
At the top of nvda.pyw:
* import _ctypes
* make a subclass of _ctypes.COMError. User-defined subclasses do have overridable attributes.
* Replace _ctypes.COMError with the subclass.

This did then allow watchdog to override the constructor later, but then all COMErrors in NVDA were no longer being caught. It looks as if the internal code that throws COMError from ctypes cfunction calls is still using the original COMError -- like it already had a reference to it and did not look it up on _ctypes when it needed it.
Do you have any cool ideas @jcsteh  of how we could restructure watchdog a bit not to have to rely on monkeypatching COMError?

**Comment by leonardder** [comment]
I guess you could get a traceback or stack trace of the way COMError is called, so you can find out what ctypes callback calls it. May be it can be monkeypatched instead?

**Comment by leonardder** [comment]
Fixed in #9795

**Comment by leonardder** [comment]
This hasn't been fixed completely, see #10276
